sql-plugin/src/main/scala/org/apache/spark/sql/rapids/collectionOperations.scala

```diff
@@ -1172,11 +1175,16 @@ case class GpuMapFromArrays(left: Expression, right: Expression) extends GpuBina
   }
 
   def compareOffsets(lhs: ColumnVector, rhs: ColumnVector) : Boolean = {
-    withResource(lhs.getListOffsetsView) { lhsOffsets =>
+    val boolScalar = withResource(lhs.getListOffsetsView) { lhsOffsets =>
       withResource(rhs.getListOffsetsView) { rhsOffsets =>
-        lhsOffsets.equalTo(rhsOffsets).all.getBoolean
+        withResource(lhsOffsets.equalTo(rhsOffsets)) { compareOffsets =>
+          compareOffsets.all
+        }
       }
     }
+    withResource(boolScalar) { all =>
+      all.getBoolean
```

Comment:
You should also check to see if it is valid first? In CUDF if all of the input values are null, then the output is null instead of false like you would expect.

Reply:
I've replaced `equalTo` with `equalToNullAware` that is guaranteed to return non NULL values. Please resolve this thread if no other fixes are required. Thanks!